core/tx_pool.go

```diff
@@ -667,9 +685,13 @@ func (pool *TxPool) enqueueTx(hash common.Hash, tx *types.Transaction) (bool, er
 		// Nothing was replaced, bump the queued counter
 		queuedGauge.Inc(1)
 	}
-	if pool.all.Get(hash) == nil {
-		pool.all.Add(tx)
-		pool.priced.Put(tx)
+	// If the transaction isn't in lookup set but it's expected to be there, boom
+	if pool.all.Get(hash) == nil && !addAll {
+		panic(fmt.Sprintf("missing transaction %s in lookup set", hash))
```

Comment:
This is fine for testing, but let's not have a `panic` here when we merge this to `master`

Reply:
The scenario is: when we promote a transaction or demote a transaction, the transaction should already exists in the lookup set. If it's not, maybe somewhere is wrong. Should I degrate it to an `Error` message?